src/main/groovy/lang/ObjectRange.java

```diff
@@ -37,7 +37,7 @@
  * Note: This class is similar to {@link IntRange}. If you make any changes to this
  * class, you might consider making parallel changes to {@link IntRange}.
  */
-public class ObjectRange extends AbstractList implements Range {
+public class ObjectRange extends AbstractList<Comparable> implements Range<Comparable> {
```

Comment:
The main change is that you use T instead of Comparable. In case of `0.1..0.9` T is supposed to be BigDecimal and something like `'a'...100` is not legal from the Java side. it will still work, because ScriptBytecodeAdapter does not care about the static type at all. But if somebody uses the ObjectRange from Java, then why not generify it in that manner for those people?